live_tests/modules/test_texter.py

```diff
@@ -0,0 +1,162 @@
+        inbox_dict = Texter().read_inbox()
+        self.assertIsInstance(inbox_dict, dict)
+
+    def test_send(self):
+        t = Texter()
+        logging.info("sending text")
+        send_status = t.send(message="Testing example message in Texter.",
+                             phone_number=TEXTLOCAL_PHONENUMBER)
+        logging.info("sleeping three minutes before reading the text")
+        time.sleep(180)
+        logging.info("reading text")
+        new_message_dict = t.read_inbox()
+        self.assertTrue("Testing example message in Texter." in new_message_dict[0])
+
+
+    def test_send_hindi(self):
+        t = Texter()
+        logging.info("sending Hindi text")
+        hindi_message = hindi_remind() + ' ' + msg_placeholder_child('Hindi')
+        send_status = t.send(message=hindi_message,
+                             phone_number=TEXTLOCAL_PHONENUMBER)
+        logging.info("sleeping three minutes before reading the text")
+        time.sleep(180)
+        logging.info("reading text")
+        new_message_dict = t.read_inbox()
+        self.assertTrue(hindi_message in new_message_dict[0])
+
+
+    def test_full_e2e_english_signup_and_cancel_flow(self):
+        logging.info("running e2e full flow for sign up + cancel in English...")
+        self.run_flow_for_language(language="English",
+                                   person_name="Testjohnson",
+                                   join_keyword="JOIN")
+
+    def test_full_e2e_hindi_signup_and_cancel_flow(self):
+        logging.info("running e2e full flow for sign up + cancel in Hindi...")
+        self.run_flow_for_language(language="Hindi",
+                                   person_name=u'\u0906\u0930\u0935',
+                                   join_keyword=hindi_remind())
+
+
+    def run_flow_for_language(self, language, person_name, join_keyword):
+        logging.info("sending text")
+        subscribe_date = datetime(2017, 1, 30).date()
+        join_text = join_keyword + " " + person_name + " " + date_to_date_string(subscribe_date)
+        t = Texter()
+        send_status = t.send(message=join_text,
+                             phone_number=TEXTLOCAL_PHONENUMBER)
+        self.assertTrue('SuccessFully' in send_status[0]['responseCode'])
+        logging.info("sleeping three minutes before reading the text")
+        time.sleep(180)
+        logging.info("reading text")
+        new_messages = t.read_inbox()[0]  # TODO: Match to job
+        tp = TextProcessor(TEXTLOCAL_PHONENUMBER)
+        processed = False
+        logging.info("checking text can be processed")
+        for text in new_messages:
+            if text == join_text:
+                processed = True
+                tp.process(text)
+        self.assertTrue(processed)
+
+        logging.info("checking contact object is created")
+        self.assertTrue(Contact.objects.filter(name=person_name,
+                                               phone_number=TEXTLOCAL_PHONENUMBER).exists())
```

Comment:
TextLocal mangles sender IDs, so I think this would also return a contact if the message in fact had been sent by a different sender ID. Though I'm not sure there's something we can really do to test this since all the IDs would show up with the same phone_number.